samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinatorFactory.java

```diff
@@ -37,20 +38,18 @@
   private static final String DEFAULT_JOB_NAME = "defaultJob";
   private static final String PROTOCOL_VERSION = "2.0";
 
-  /**
-   * Instantiates an {@link ZkJobCoordinator} using the {@link Config}.
-   *
-   * @param config zookeeper configurations required for instantiating {@link ZkJobCoordinator}
-   * @return An instance of {@link ZkJobCoordinator}
-   */
   @Override
-  public JobCoordinator getJobCoordinator(String processorId, Config config, MetricsRegistry metricsRegistry) {
+  public JobCoordinator getJobCoordinator(String processorId, Config config, MetricsRegistry metricsRegistry,
+      MetadataStore coordinatorStreamStore) {
     // TODO: Separate JC related configs into a "ZkJobCoordinatorConfig"
     String jobCoordinatorZkBasePath = getJobCoordinationZkPath(config);
     ZkUtils zkUtils = getZkUtils(config, metricsRegistry, jobCoordinatorZkBasePath);
     LOG.debug("Creating ZkJobCoordinator with config: {}.", config);
-    ZkMetadataStore metadataStore = new ZkMetadataStore(zkUtils.getKeyBuilder().getRootPath(), config, metricsRegistry);
-    return new ZkJobCoordinator(processorId, config, metricsRegistry, zkUtils, metadataStore);
+
+    // TODO: This should be merged with coordinatorStreamStore
```

Comment:
Would be a good idea to create a JIRA for the above TODO and link it here. Please assign the JIRA to rightful owner.

Reply:
Sure will do.